nvflare/private/fed/client/client_executor.py

```diff
@@ -214,6 +185,18 @@ def start_app(
         )
         thread.start()
 
+    def _get_job_launcher(self, job_id, client, job_meta: dict) -> dict:
+        launch_image = extract_job_image(job_meta, client.client_name)
+        launch_data = {JobConstants.JOB_IMAGE: launch_image, JobConstants.JOB_ID: job_id}
+
+        job_launcher = None
+        for _, component in client.components.items():
+            if isinstance(component, JobLauncherSpec):
+                if component.can_launch(launch_data):
+                    job_launcher = component
+        launch_data[JobConstants.JOB_LAUNCHER] = job_launcher
```

Comment:
Once event is fired, we just need to decide which launcher to use. There can be 0, 1, or more launchers that said they can handle the job.

Reply:
changed to use event firing to get job launcher.